### PR TITLE
feat: enables dependabot

### LINKED ISSUE #321: enable dependabot

### PR DESCRIPTION
**Pull request by nilekhc**
**What this PR does / why we need it**:
This PR enables dependabot to update Gomod and GitHub action dependencies.

**Which issue(s) does this PR fix** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #321

**Special notes for your reviewer**:

### CODE REVIEW

File: .github/dependabot.yml
Comment (apeabody) [inline .github/dependabot.yml:20]: You might want to include `yarn`? (untested example)

```suggestion
      prefix: "chore"

  - package-ecosystem: "npm"
    directory: "/"
    schedule:
      interval: "weekly"
    commit-message:
      prefix: "chore"
```

**Comment by sozercan** [comment]
i think this is duplicate of #320

**Comment by nilekhc** [comment]
> i think this is duplicate of #320

Ah I see. But this does not have `github-actions` package-system enabled. How do we do this?

**Review by apeabody — APPROVED** [review]
LGTM! Included a suggestion

File: .github/dependabot.yml
Comment (apeabody) [reply .github/dependabot.yml:20]: I just saw the comment about re: #320, which does have "npm" for yarn.

**Comment by apeabody** [comment]
> > i think this is duplicate of #320
> 
> Ah I see. But this does not have `github-actions` package-system enabled. How do we do this?

One option would be to re-scope this PR to just `github-actions` and after committed merge main into #320, or use this PR to add `github-actions` after #320 is committed.

**Comment by nilekhc** [comment]
> > > i think this is duplicate of #320
> > 
> > 
> > Ah I see. But this does not have `github-actions` package-system enabled. How do we do this?
> 
> One option would be to re-scope this PR to just `github-actions` and after committed merge main into #320, or use this PR to add `github-actions` after #320 is committed.

Yeah. I updated the title so we don't merge accidentally. I'll revisit this after #320 is merged.

**Comment by sozercan** [comment]
merged #320

**Comment by nilekhc** [comment]
@apeabody @sozercan I have updated the config file to:
- change frequency to weekly
- limit version upgrade to `patch` for Go and NPM

Let me know wdyt.

**Comment by apeabody** [comment]
> @apeabody @sozercan I have updated the config file to:
> 
> * change frequency to weekly
> * limit version upgrade to `patch` for Go and NPM
> 
> Let me know wdyt.

I think limiting to patch should be OK as I recall Dependabot will open for security updates regardless of the `update-types` condition.

File: .github/dependabot.yml
Comment (apeabody) [inline .github/dependabot.yml:78]:
```suggestion
    directory: /.github
```

**Review by apeabody — APPROVED** [review]
LGTM! Suggestion on the `github-actions` path below

File: .github/dependabot.yml
Comment (sozercan) [reply .github/dependabot.yml:78]: this is interesting, we only have `/` in GK but definately get GHA updates

Comment (sozercan) [reply .github/dependabot.yml:78]: i am guessing `.github/workflows` is built-in since actions must live there

File: .github/dependabot.yml
Comment (sozercan) [inline .github/dependabot.yml:75]:
```suggestion
        - "version-update:semver-minor"
```

File: .github/dependabot.yml
Comment (nilekhc) [reply .github/dependabot.yml:78]: Yeah, we also have `/` on other repos which works just fine. I'll keep this as is since it's tested.

**Review by apeabody — APPROVED** [review]
LGTM!

**Pull request merged**
